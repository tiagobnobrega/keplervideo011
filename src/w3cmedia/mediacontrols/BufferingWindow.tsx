/*
 * Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All rights reserved.
 *
 * AMAZON PROPRIETARY/CONFIDENTIAL
 *
 * You may not use this file except in compliance with the terms and
 * conditions set forth in the accompanying LICENSE.TXT file.
 *
 * THESE MATERIALS ARE PROVIDED ON AN "AS IS" BASIS. AMAZON SPECIFICALLY
 * DISCLAIMS, WITH RESPECT TO THESE MATERIALS, ALL WARRANTIES, EXPRESS,
 * IMPLIED, OR STATUTORY, INCLUDING THE IMPLIED WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE, AND NON-INFRINGEMENT.
 */

import LottieView from '@amzn/lottie-react-native';
import React from 'react';
import { StyleSheet, View, ViewProps } from 'react-native';
import { COLORS } from '../../styles/Colors';
import { scaleUxToDp } from '../../utils/pixelUtils';

const BufferingWindow = React.memo((props: ViewProps) => {
  return (
    <View {...props} style={styles.container} testID="buffering-view">
      <LottieView
        testID="video-buffer-view"
        source={require('../../js/resources/loader.json')}
        autoPlay
        loop
        style={styles.loader}
      />
    </View>
  );
});

export const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    top: 0,
    left: 0,
    position: 'absolute',
    flex: 1,
    backgroundColor: COLORS.BLACK,
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    zIndex: 5,
  },
  loader: {
    width: scaleUxToDp(80),
    height: scaleUxToDp(320),
    alignSelf: 'center',
  },
});
export default BufferingWindow;
