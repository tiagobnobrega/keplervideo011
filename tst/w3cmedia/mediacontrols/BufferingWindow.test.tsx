import LottieView from '@amzn/lottie-react-native';
import { describe } from '@jest/globals';
import { render } from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import BufferingWindow, {
  styles,
} from '../../../src/w3cmedia/mediacontrols/BufferingWindow';

describe('BufferingWindow Test Cases', () => {
  it('renders correctly', () => {
    const tree = render(<BufferingWindow />);
    expect(tree).toMatchSnapshot();
  });

  it('View is present in component', async () => {
    const { queryByTestId } = render(<BufferingWindow />);
    const view = queryByTestId('buffering-view');
    expect(view).toBeDefined();
  });

  it('LottieView is present in component with correct props', async () => {
    const { UNSAFE_queryAllByType } = render(<BufferingWindow />);
    const lottieView = UNSAFE_queryAllByType(LottieView);
    expect(lottieView).toBeDefined();
    const lottieViewWithStyles = lottieView.find(
      lottie => lottie.props.style === styles.loader,
    );
    expect(lottieViewWithStyles).toBeTruthy();
    const lottieViewWithSource = lottieView.find(
      lottie =>
        lottie.props.source ===
        require('../../../src/js/resources/loader.json'),
    );
    expect(lottieViewWithSource).toBeTruthy();
    const lottieViewWithLoop = lottieView.find(
      lottie => lottie.props.loop === true,
    );
    expect(lottieViewWithLoop).toBeTruthy();
  });
});
