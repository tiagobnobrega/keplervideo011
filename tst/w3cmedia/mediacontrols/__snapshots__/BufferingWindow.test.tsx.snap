// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BufferingWindow Test Cases renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 5,
    }
  }
  testID="buffering-view"
/>
`;
